### PR TITLE
Build PT1.9 inference

### PR DESCRIPTION
**Pull request by tejaschumbalkar**
*GitHub Issue #, if available:*

Note: If merging this PR should also close the associated Issue, please also add that Issue # to the Linked Issues section on the right. 

### Description
Build PT1.9 inference
### Tests run
**NOTE: If you are creating a PR for a new framework version, please ensure success of the standard, rc, and efa sagemaker remote tests by updating the dlc_developer_config.toml file:**

- [ ] Revision A: `sagemaker_remote_tests = "standard"`
- [ ] Revision B: `sagemaker_remote_tests = "rc"`
- [ ] Revision C: `sagemaker_remote_tests = "efa"`

**Additionally, please run the sagemaker local tests in at least one revision:**
- [ ] `sagemaker_local_tests = true`

## Label Checklist
- [ ] I have added the project label for this PR (*<project_name>* or "Improvement")

## PR Checklist
- [ ] I've prepended PR tag with frameworks/job this applies to : [mxnet, tensorflow, pytorch] | [ei/neuron/graviton] | [build] | [test] | [benchmark] | [ec2, ecs, eks, sagemaker]
- [ ] If the PR changes affects SM test, I've modified dlc_developer_config.toml in my PR branch by setting sagemaker_tests = true and efa_tests = true
- [ ] If this PR changes existing code, the change fully backward compatible with pre-existing code. (Non backward-compatible changes need special approval.)
- [ ] (If applicable) I've documented below the DLC image/dockerfile this relates to
- [ ] (If applicable) I've documented below the tests I've run on the DLC image
- [ ] (If applicable) I've reviewed the licenses of updated and new binaries and their dependencies to make sure all licenses are on the Apache Software Foundation Third Party License Policy Category A or Category B license list.  See [https://www.apache.org/legal/resolved.html](https://www.apache.org/legal/resolved.html).
- [ ] (If applicable) I've scanned the updated and new binaries to make sure they do not have vulnerabilities associated with them.

## Pytest Marker Checklist
- [ ] (If applicable) I have added the marker `@pytest.mark.model("<model-type>")` to the new tests which I have added, to specify the Deep Learning model that is used in the test (use `"N/A"` if the test doesn't use a model)
- [ ] (If applicable) I have added the marker `@pytest.mark.integration("<feature-being-tested>")` to the new tests which I have added, to specify the feature that will be tested
- [ ] (If applicable) I have added the marker `@pytest.mark.multinode(<integer-num-nodes>)` to the new tests which I have added, to specify the number of nodes used on a multi-node test
- [ ] (If applicable) I have added the marker `@pytest.mark.processor(<"cpu"/"gpu"/"eia"/"neuron">)` to the new tests which I have added, if a test is specifically applicable to only one processor type

#### EIA/NEURON/GRAVITON Testing Checklist
* When creating a PR:
- [ ] I've modified `dlc_developer_config.toml` in my PR branch by setting `ei_mode = true`, `neuron_mode = true` or `graviton_mode = true`

#### Benchmark Testing Checklist
* When creating a PR:
- [ ] I've modified `dlc_developer_config.toml` in my PR branch by setting `benchmark_mode = true`

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license. I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: pytorch/inference/docker/1.9/py3/Dockerfile.cpu [hunk 1]
```diff
@@ -108,7 +108,7 @@ RUN conda install -c \
  && ln -s /opt/conda/bin/pip /usr/local/bin/pip3 \
  && pip install packaging==20.4 \
     enum-compat==0.0.3 \
-    numpy==1.20.3 \
+    numpy>=1.22.2 \
```
Comment (arjkesh) [inline pytorch/inference/docker/1.9/py3/Dockerfile.cpu:111]: this needs to be in quotations due to the >= operator

File: pytorch/inference/docker/1.9/py3/cu111/Dockerfile.gpu [hunk 2]
```diff
@@ -166,10 +166,12 @@ RUN pip uninstall -y torch \
 
 RUN conda install -y -c conda-forge "pyyaml>=5.4,<5.5"
 RUN pip install \
-    "Pillow>=8.3.2,<8.4" \
-    "awscli<2"
+    "Pillow>=9.0.1" \
+    "awscli<2" \
+    "urllib3>=1.26.5" \
+    mkl
```
Comment (arjkesh) [inline pytorch/inference/docker/1.9/py3/cu111/Dockerfile.gpu:172]: why is MKL installed here?

File: pytorch/inference/docker/1.9/py3/cu111/Dockerfile.gpu [hunk 3]
```diff
@@ -166,10 +166,12 @@ RUN pip uninstall -y torch \
...
 
-RUN pip install --no-cache-dir "sagemaker-pytorch-inference>=2"
+RUN pip install --no-cache-dir "sagemaker-pytorch-inference==2.0.6"
```
Comment (tejaschumbalkar) [inline pytorch/inference/docker/1.9/py3/cu111/Dockerfile.gpu:174]: Are we pinning explicitly for compatibility with the framework version?

File: test/dlc_tests/ec2/pytorch/inference/test_pytorch_inference.py [hunk 4]
```diff
@@ -87,8 +87,8 @@ def test_ec2_pytorch_inference_eia_gpu(pytorch_inference_eia, ec2_connection, re
 @pytest.mark.parametrize("ec2_instance_type", PT_EC2_GPU_INSTANCE_TYPE, indirect=True)
 def test_pytorch_inference_torchaudio_gpu(pytorch_inference, ec2_connection, gpu_only, ec2_instance_type):
     _, image_framework_version = get_framework_and_version_from_tag(pytorch_inference)
-    if Version(image_framework_version) not in SpecifierSet(">=1.9,!=1.10.*"):
-        pytest.skip("torchaudio is installed in PT 1.9 and above, except PT 1.10.*")
+    if Version(image_framework_version) not in SpecifierSet("!=1.9.*,!=1.10.*"):
+        pytest.skip("torchaudio is not supported in PT 1.9 and 1.10")
```
Comment (tejaschumbalkar) [inline test/dlc_tests/ec2/pytorch/inference/test_pytorch_inference.py:91]: I do see torchaudio installed in PT1.9 dockerfiles?

Comment (arjkesh) [reply test/dlc_tests/ec2/pytorch/inference/test_pytorch_inference.py:91]: @waytrue17 do you know why we are skipping PT 1.9 torchaudio?

File: pytorch/inference/docker/1.9/py3/cu111/Dockerfile.gpu (hunk 3 above)
Comment (tejaschumbalkar) [reply pytorch/inference/docker/1.9/py3/cu111/Dockerfile.gpu:174]: pinned the version similar to the prod image

File: pytorch/inference/docker/1.9/py3/cu111/Dockerfile.gpu (hunk 2 above)
Comment (ydaiming) [reply pytorch/inference/docker/1.9/py3/cu111/Dockerfile.gpu:173]: PyTorch requires MKL: https://github.com/pytorch/pytorch#install-dependencies
It's included. I think the error comes from version mismatch. Is the error still occurring now?

File: test/dlc_tests/ec2/pytorch/inference/test_pytorch_inference.py (hunk 4 above)
Comment (ydaiming) [reply test/dlc_tests/ec2/pytorch/inference/test_pytorch_inference.py:91]: Because it didn't really work.

**Review by arjkesh — APPROVED** [review]

**Pull request merged**